src/presenter/trip-point-board-presenter.js

```diff
@@ -1,60 +1,151 @@
 import { render, remove } from '../framework/render';
-import { FilterType, SortType } from '../const';
-import { filterListFuture, filterListPresent, filterListPast, sortListDay, sortListPrice, sortListTime } from '../utils';
+import { UserAction, UpdateType } from '../const';
+
 
 import TripPointPresenter from '../presenter/trip-point-presenter';
 
+import AddNewTripButtonView from '../view/add-new-trip-button-view';
+
+
 import TripPointZeroView from '../view/zero-point-view';
 
 
 export default class TripPointBoardPresenter{
   #tripEventsListContainer = null;
+  #tripHeaderContainer = null;
 
   #listPresernter = new Map();
-  #zeroPointsPresenter = null;
+
+  #zeroPointsView = null;
+  #addNewTripButtonView = null;
 
   #tripPointsModel = null;
   #offersModel = null;
   #destinationsModel = null;
 
-  #points = null;
+  #filterModel = null;
+  #sortModel = null;
+
+  #filterPresenter = null;
+
   #offers = null;
   #destinations = null;
 
+  #currentNewPoint = null;
 
   constructor(
     {
+      tripHeaderContainer,
       tripEventsListContainer,
+
       tripPointsModel,
       offersModel,
-      destinationsModel
+      destinationsModel,
+
+      filterModel,
+      sortModel,
+
+      filterPresenter
     }
   ){
     this.#tripEventsListContainer = tripEventsListContainer;
+    this.#tripHeaderContainer = tripHeaderContainer;
 
     this.#tripPointsModel = tripPointsModel;
     this.#offersModel = offersModel;
     this.#destinationsModel = destinationsModel;
 
-    this.#points = this.#tripPointsModel.points;
+    this.#filterModel = filterModel;
+    this.#sortModel = sortModel;
+
+    this.#filterPresenter = filterPresenter;
+
     this.#offers = this.#offersModel.offers;
     this.#destinations = this.#destinationsModel.destinations;
 
-    this.#zeroPointsPresenter = new TripPointZeroView();
+    this.#tripPointsModel.addObserver(this.#handleModelEvent);
+    this.#sortModel.addObserver(this.#handleModelEvent);
+    this.#filterModel.addObserver(this.#handleModelEvent);
+
+    this.#addNewTripButtonView = new AddNewTripButtonView({newPoinHandler:this.#newPoinHandler});
```

Comment:
Наблюдаю небольшой рассинхрон в именовании хэндлеров, думаю лучше будет примерно так:
```suggestion
    this.#addNewTripButtonView = new AddNewTripButtonView({ handleAddNewPoin: this.#newPoinHandler});
```